### DISCUSSION TITLE
How to write a style tag

### DISCUSSION THREAD

**Discussion by onichandame** [DioxusLabs/dioxus#753] [Q&A] [answered]
I wish to write some css in style tag. I saw https://github.com/DioxusLabs/dioxus/issues/68#issuecomment-1007873515 but it didn't help.

![image](https://user-images.githubusercontent.com/23728505/211830512-8ce6dc9e-bba1-45fb-90b9-dcb9d799a5b0.png)

My question is, what is the intended way to add style blocks?

**Comment by ealmloff** [comment, marked as answer]
You need to call into_iter() and use rsx to format the text:
```rust

pub fn App(cx: Scope) -> Element {
    render! {
        div {
            p {
                "Hello!"
            }
            style {
                vec!["p { color: red; }"].into_iter().map(|s| rsx!{s})
            }
        }
    }
}
```

**Reply by onichandame** [reply]
It would be nice to have the boilerplate handled automatically by the macro tho
